RecoTracker/TkTrackingRegions/src/OuterHitPhiPrediction.h

```diff
@@ -4,32 +4,40 @@
 /** predicts phi range at a given radius r */
 
 #include "RecoTracker/TkMSParametrization/interface/PixelRecoRange.h"
-#include "RecoTracker/TkTrackingRegions/interface/TkTrackingRegionsMargin.h"
 
 #include "FWCore/Utilities/interface/Visibility.h"
 
+#include <cassert>
+
 class dso_internal OuterHitPhiPrediction {
 public:
 
-  typedef PixelRecoRange<float> Range;
-  typedef TkTrackingRegionsMargin<float> Margin;
+  using Range= PixelRecoRange<float>;
 
   OuterHitPhiPrediction( 
       const Range & phiAtVertex, 
       const Range & curvature, 
-      float originRBound, 
-      const Margin & tolerance = Margin(0,0))
+      float originRBound) 
     : thePhiAtVertex(phiAtVertex), theCurvature(curvature),
-      theOriginRBound (originRBound), theTolerance(tolerance) { } 
+      theOriginRBound (originRBound) {
+       assert(theCurvature.max()>0);
```

Comment:
@Martin-Grunewald I just realized that you are trying to use a fixed snapshot time Global Tag on data taken in the relative future with respect to the snapshot time itself.
The GT `80X_dataRun2_HLT_frozen_v12` has snapshot set to:  2016-04-20 06:59:54 UTC.
The run you are using was taken on: 2016.05.06 22:27:33 UTC.
This simply means that you are seeing plain wrong conditions for that IOV!
As an example, the last IOV for  `RunInfoRcd` (that drives the magnetic field map you load) seen in the GT `80X_dataRun2_HLT_frozen_v12` is at run 270395, 

https://cmswbm.web.cern.ch/cmswbm/cmsdb/servlet/RunSummary?RUN=270395&SUBMIT=Submit

which happens to be a 0T run !
And indeed if you dump the magnet currents you'll get:
```
$ conddb --noLimit list runinfo_start_31X_hlt | grep 270395
270395       2016-04-20 06:32:39  786aaad4b7bcfab74d0d10fde752e22b637168bf  RunInfo      
$ conddb dump 786aaad4b7bcfab74d0d10fde752e22b637168bf | grep current
	<m-start-current>1.063721371e+01</m-start-current>
	<m-stop-current>1.063721371e+01</m-stop-current>
	<m-avg-current>1.063721371e+01</m-avg-current>
	<m-max-current>1.063721371e+01</m-max-current>
	<m-min-current>1.063721371e+01</m-min-current>
```
while it should be:
```
$ conddb --noLimit list runinfo_start_31X_hlt | grep 272762
272762       2016-05-06 22:27:38  bcef1447160d669c62328b485ce1cfdd4c4a4efd  RunInfo      
$ conddb dump bcef1447160d669c62328b485ce1cfdd4c4a4efd > dump_bcef1447160d669c62328b485ce1cfdd4c4a4efd.xml | grep current
	<m-start-current>1.814441211e+04</m-start-current>
	<m-stop-current>1.814441211e+04</m-stop-current>
	<m-avg-current>1.814441211e+04</m-avg-current>
	<m-max-current>1.814441211e+04</m-max-current>
	<m-min-current>1.814441211e+04</m-min-current>
```
Reconstructing a 3.8T run with 0T conditions is bound to fail, IMHO.
Frozen GTs are meant for relvals and there is **NO** guarantee at all they are going to cover anything else outside run ranges covered in relvals. If you wish to test an arbitrary run please use the open snapshot GTs made available here:

https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideFrontierConditions#HLT_reconstruction_Global_Tags

otherwise wait for 2016 data relvals to become available.